src/controller/cve-id.controller/cve-id.controller.js

```diff
@@ -240,184 +246,262 @@ async function modifyCveId (req, res) {
-    requested_by: {
-      cna: 'orgshortname',
-      user: 'you'
-    },
-    time: {
-      created: '3000-06-24T21:08:42.937+00:00',
-      modified: '3000-06-24T21:12:46.576+00:00',
-      reserved: '3000-06-24T21:12:46.576+00:00'
+  }
+}
+
+async function priorityReservation (year, amount, cveId, shortName, cnaShortName, requester, availableIds, res) {
+  let result = await CveIdRange.findOne({ cve_year: year })
+  let returned = false
+  let isFull = false
+
+  // Cve Id Range for 'year' does not exists
+  if (!result) {
+    returned = true
+    logger.info('CVE IDs for year ' + year + ' cannot be reserved at this time.')
+    return res.status(403).json({ message: 'CVE IDs for year ' + year + ' cannot be reserved at this time.' })
+  }
+
+  if (!returned) {
+    const endRange = parseInt(result.ranges.priority.end)
+    result = await CveIdRange.findOneAndUpdate({ $and: [{ cve_year: year }, { 'ranges.priority.top_id': { $lt: endRange } }] }, { $inc: { 'ranges.priority.top_id': amount } }, { new: true })
+
+    // priority id block is full, reserve id in sequential block
+    if (!result) {
+      isFull = true
+      logger.info('Priority id block is full for year ' + year + ', reserving id in sequential block.')
+      await sequentialReservation(year, amount, cveId, shortName, cnaShortName, requester, availableIds, true, res)
     }
-  },
-  {
-    cve_id: 'CVE-3000-0518',
-    cve_year: 3000,
-    owning_cna: 'orgshortname',
-    state: 'RESERVED',
-    requested_by: {
-      cna: 'orgshortname',
-      user: 'you'
-    },
-    time: {
-      created: '3000-06-24T21:08:42.937+00:00',
-      modified: '3000-06-24T21:12:46.576+00:00',
-      reserved: '3000-06-24T21:12:46.576+00:00'
+  }
+
+  if (!isFull && !returned) {
+    const id = generateIds(year, result.ranges.priority, amount)
+
+    cveId = new CveId()
+    cveId.cve_id = id[0]
+    cveId.cve_year = year
+    cveId.state = 'RESERVED'
+    cveId.owning_cna = shortName // the org who gets assigned the reserved CVE IDs
+    cveId.reserved = Date.now()
+    cveId.requested_by = {
+      cna: cnaShortName, // the org who requested the CVE IDs
+      user: requester
     }
-  },
-  {
-    cve_id: 'CVE-3000-0591',
-    cve_year: 3000,
-    owning_cna: 'orgshortname',
-    state: 'RESERVED',
-    requested_by: {
-      cna: 'orgshortname',
-      user: 'you'
-    },
-    time: {
-      created: '3000-06-24T21:08:42.937+00:00',
-      modified: '3000-06-24T21:12:46.576+00:00',
-      reserved: '3000-06-24T21:12:46.576+00:00'
+
+    // Create a CveId document (reserve the CVE ID)
+    await CveId.findOneAndUpdate().byCveId(id).updateOne(cveId).setOptions({ upsert: true })
+
+    res.header('CVE-API-REMAINING-QUOTA', availableIds - amount)
+    logger.info({ message: 'A priority CVE ID was reserved for \'' + shortName + '\' org on behalf of \'' + cnaShortName + '\' org.', cve_id: id })
+    return res.status(200).json({ message: 'The following CVE IDs were reserved.', reserved: id })
```

Comment:
This should return an object with a cve_ids field that's an array of cve ids. I don't think we need the message.